src/components/Contact.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import InLineSpinner from './InLineSpinner';
+import Avatar from './Avatar';
+import Button from './Button';
+
+class Contact extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { contact: null };
+  }
+
+  componentDidMount() {
+    const { contact } = this.props;
```

Comment:
You don't need to add the contact to component state. You can use it directly from props.